### ISSUE TITLE
Window function's output order is wrong

### ISSUE THREAD

**Issue by winoros** [pingcap/tidb#9806]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
CREATE TABLE `t1` (
  `id` int(11) DEFAULT NULL,
  `sex` char(1) DEFAULT NULL
);
insert into t1 values(1, 'F'), (2, 'M');
SELECT NTILE(2) OVER (ORDER BY sex) FROM t1 ORDER BY id+2 DESC;
```

2. What did you expect to see?

execute successfully.

3. What did you see instead?

panicked

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

**Comment by winoros** [comment]
After some investigation, when reading Int value, the column reached to places where stores Char.

**Comment by zz-jason** [comment]
Please also post the panic stack in the `What did you see instead` section

**Comment by winoros** [comment]
@zz-jason 
```
[2019/03/19 16:53:50.686 +08:00] [WARN] [builder.go:1095] ["build proj exec"] ["column indices"="[0 1 2]"] [columns="[test.t1.sex test.t1.id 5_window_3]"]
[2019/03/19 16:53:50.686 +08:00] [WARN] [builder.go:1095] ["build proj exec"] ["column indices"="[2]"] [columns="[ntile(2) over (order by sex)]"]
[2019/03/19 16:53:50.687 +08:00] [WARN] [row.go:58] ["column data"] ["bytes data"="[70 70 70 77 77]"] ["byte len"=5] ["self len"=7] ["row num"=7] ["all col byte len"="[56,5,56]"]
```
From the schema, the three column of the projection should be `char, int, int`. But from the actual data in chunk, the order is `int, char, int`.
The cause of panic is the `output wrong order`. So panic stack is useless.

**Comment by winoros** [comment]
After checking, fixed by #10452
